### PR TITLE
Standard library documentation continued.

### PR DESCRIPTION
**Pull request by yole**

### CODE REVIEW

File: core/builtins/native/kotlin/Collections.kt [hunk 1]
```diff
@@ -16,40 +16,112 @@
 
 package kotlin
 
+/**
+ * Classes that inherit from this trait can be represented as a sequence of elements that can
+ * be iterated over.
+ */
 public trait Iterable<out T> {
```
Comment (orangy) [inline core/builtins/native/kotlin/Collections.kt]: doc for `T`?

File: core/builtins/native/kotlin/Collections.kt [hunk 2]
```diff
@@ -126,32 +254,100 @@ public trait MutableSet<E> : Set<E>, MutableCollection<E> {
...
+ * A collection that holds pairs of objects (keys and values) and supports efficiently retrieving
+ * the value corresponding to each key. Map keys are unique; the map holds only one value for each key.
+ * Methods in this trait support only read-only access to the map; read-write access is supported through
+ * the [MutableMap] trait.
+ */
 public trait Map<K, out V> {
     // Query Operations
+    /**
+     * Returns the size of the map.
+     */
```
Comment (orangy) [inline core/builtins/native/kotlin/Collections.kt]: size is a little bit unclear, may be "number of key-value" pairs?

File: core/builtins/native/kotlin/Collections.kt [hunk 3]
```diff
@@ -126,32 +254,100 @@ public trait MutableSet<E> : Set<E>, MutableCollection<E> {
...
     // Views
+    /**
+     * Returns a [Set] of all keys in this map.
+     */
     public fun keySet(): Set<K>
+
+    /**
+     * Returns a [Collection] of all values in this map.
+     */
     public fun values(): Collection<V>
```
Comment (orangy) [inline core/builtins/native/kotlin/Collections.kt]: May contain duplicate values

**Comment by yole** [comment]
Merged manually.

**Pull request closed without merging**